lib/explorer/backend/data_frame.ex

```diff
@@ -154,8 +154,13 @@ defmodule Explorer.Backend.DataFrame do
   @callback collect(df) :: df
   @callback from_tabular(Table.Reader.t(), dtypes) :: df
   @callback from_series([{binary(), Series.t()}]) :: df
-  @callback to_rows(df, atom_keys? :: boolean()) :: [map()]
-  @callback to_rows_stream(df, atom_keys? :: boolean(), chunk_size :: integer()) :: Enumerable.t()
+  @callback to_rows(df, atom_keys? :: boolean(), names :: [column_name()]) :: [map()]
+  @callback to_rows_stream(
+              df,
+              atom_keys? :: boolean(),
+              chunk_size :: integer(),
+              names :: [column_name()]
+            ) :: Enumerable.t()
```

Comment:
Isn't the names argument here equivalent to doing a select before?

Reply:
I think we can remove the argument here and do a DF.select inside DF.transform.

Reply:
`DF.select` inside `DF.transform` is how it was before. I liked the idea of moving it to `DF.to_rows` and `DF.to_rows_stream` because:

1. Those functions become more expressive.
2. `DF.transform` also becomes more performant.

Right now the `to_rows` functions zip column-by-column. If you simply tell them to serialize fewer columns, we achieve the same thing but without the extra `select`.

Reply:
> DF.transform also becomes more performant.

How does it become more performant? Select is a very cheap operation on the dataframes side. The cost of serialization will easily trump anything else?

The downside is having more ways of achieving the same thing, larger API surfaces for adapters to implement, and so on. While the differences are minimal:

```
df |> DF.select([:foo, :bar]) |> DF.to_rows()
df |> DF.to_rows(names: [:foo, :bar])
```

Reply:
> How does it become more performant?

I just mean you need to pay the serialization cost from `to_rows` either way, so the pre-`select` doesn't buy anything if you can tell `to_rows` to serialize fewer columns.

> The downside is having more ways of achieving the same thing, larger API surfaces for adapters to implement, and so on.

But this logic is internal to `DF.transform`, no? Or do you mean adding `:names` to `to_rows` is more to maintain?

Reply:
Oh wait I'm seeing your point. Since `select` is so cheap, let's have that be _the way_. 👍

Reply:
> Oh wait I'm seeing your point. Since select is so cheap, let's have that be the way. 👍

Exactly. I think `:names` makes sense for DF.transform, because you don't want to select before, but not for the smaller ones.